### PR TITLE
Ensure we have right perms on saveconfig

### PR DESCRIPTION
**Pull request by pkalever**
Ensure to :
* set 0o600 perms on backupfiles
* set right perms on backup dir
* set right perms on /etc/target/ dir

### CODE REVIEW

File: targetcli/ui_root.py [hunk 1]
```diff
@@ -109,12 +109,21 @@ def _save_backups(self, savefile):
...
             try:
-                os.makedirs(backup_dir)
+                os.makedirs(backup_dir, mode)
             except OSError as exe:
                 raise ExecutionError("Cannot create backup directory [%s] %s."
                                      % (backup_dir, exe.strerror))
+            finally:
+                os.umask(umask_original)
+        else:
+            if (os.stat(backup_dir).st_mode & 0o777) != mode:
```
[the remark this answers is not in the document]
Comment (pkalever) [reply targetcli/ui_root.py]: saveconfig.json file will always be created 0600 every time
So no point even if the admin sets 0644 or even wider 777 on /etc/target, users cannot open saveconfig.json

**Comment by maurizio-lombardi** [comment]
Tested, looks ok to me

**Comment by carnil** [comment]
Hi, looking trough the CVE feed from MITRE, this issue seems to have been assigned [CVE-2020-13867](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-13867).

**Comment by gonzoleeman** [comment]
Note: the fix for this issue caused another issue: now targetcli always changes the directory where config files are saved to 0600, even if it is not /etc/target (e.g. /tmp, /dev, /etc).

See #198

**Pull request merged**